docs/events_handling.md

```diff
@@ -0,0 +1,89 @@
+
+# Events handling
+
+## Subscribed events
+
+Precaution needs to subscribe to a number of events when it's
+to satisfy different use cases of our users.
+
+The events can be separated into two groups `pull requests
+events` and `check suites/runs events`.
+
+### Pull requests events
+
+Official GitHub documentation: [PullRequest event](https://developer.github.com/v3/activity/events/types/#pullrequestevent)
+
+* pull_request.opened - the first event which occurs when a new pull request is created. It's a self-explanatory
+
+* pull_request.reopened - notifies Precaution to start a new scan on a reopened pull request
+
+* pull_request.synchronize - occurs when you add a commit to already opened pull request.
+We discussed it and added it in pull request number [#113](https://github.com/vmware/precaution/pull/113)
+
+### Check run/suite events
+
+Official GitHub documentation:
+- [CheckRun event](https://developer.github.com/v3/activity/events/types/#checkrunevent)
+- [CheckSuite event](https://developer.github.com/v3/activity/events/types/#checksuiteevent)
+
+You had already opened a pull request and you are at the pull request page.
+Then you navigate to the "Checks" tab.
+If the Precaution check had failed you will see "Re-run all checks", "Re-run" and
+"Re-run failed checks" buttons.
+
+* check_run.rerequested - occurs when you click the "Re-run" button.
+This will tell Precaution to start a new check run.
+
+* check_suite.rerequested - happens when you click "Re-run all checks" 
+or "Re-run failed checks" buttons. 
+This notifies Precaution and the other GitHub apps 
```

Comment:
What other GitHub apps?  The context here is unclear.

Reply:
Okay, I won’t mention the other GitHub apps at all.